### PR TITLE
feat: add tsup bundler

### PR DESCRIPTION
**Pull request by ragingwind**
## Overview

We need to change in bundler of libraries for consistency between library projects. tsup is easy to use, and is suitable to our conditions. Please check items below:

- more options for tsup config
- more deprecated packages in package.json
- more configs in package.json when it comes to publish

## PR Checklist

- [ ] I read and included theses actions below

1. I have read the [Contributing Guide](https://github.com/toss/slash/blob/main/.github/CONTRIBUTING.md)
2. I have written documents and tests, if needed.

### CODE REVIEW

File: packages/common/assert/package.json [hunk 1]
```diff
@@ -4,18 +4,17 @@
...
   },
   "main": "src/index.ts",
   "files": [
-    "dist",
-    "esm"
+    "dist"
   ],
   "scripts": {
-    "build": "rm -rf dist esm && tsc -p tsconfig.build.json --declaration --emitDeclarationOnly --declarationDir dist && rollup -c rollup.config.js",
+    "build": "rm -rf dist && tsup",
```
Comment (artechventure) [inline packages/common/assert/package.json:17]: "rm -rf" seems unnecessary when tsup option "clean: true"

File: packages/common/assert/package.json [hunk 2]
```diff
@@ -4,18 +4,17 @@
   "sideEffects": false,
   "exports": {
     ".": {
-      "require": "./src/index.ts",
-      "import": "./src/index.ts"
+      "require": "./src/index.js",
```
Comment (artechventure) [inline packages/common/assert/package.json:7]:
```suggestion
      "require": "./src/index.ts",
```

File: packages/common/assert/package.json [hunk 3]
```diff
@@ -4,18 +4,17 @@
...
+      "import": "./src/index.mjs"
```
Comment (artechventure) [inline packages/common/assert/package.json:8]:
```suggestion
      "import": "./src/index.ts"
```

File: configs/tsup/package.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@toss/tsup-config",
+  "version": "0.1.0",
+  "main": "./src/index.js",
+  "types": "./src/index.d.ts",
+  "devDependencies": {
+    "tsup": "^7.1.0"
+  },
+  "publishConfig": {
```
Comment (manudeli) [inline configs/tsup/package.json:9]: How about defining tsup v7 as peerDependency of this package

```suggestion
  "peerDependencies":{
    "tsup": "^7"
  },
  "publishConfig": {
```

**Comment by ragingwind** [comment]
sorry for confusing. tsup will be replaced  in favor of rollup

**Comment by raon0211** [comment]
Closing since this pull request is stale. Please open the pull request again if possible.

**Pull request closed without merging**